vmhost/vmhooks/bigIntOps.go

```diff
@@ -138,7 +138,11 @@ func (context *VMHooksImpl) BigIntStorageLoadUnsigned(keyOffset executor.MemPtr,
 	if context.WithFault(err, runtime.BigIntAPIErrorShouldFailExecution()) {
 		return -1
 	}
-	storage.UseGasForStorageLoad(bigIntStorageLoadUnsignedName, trieDepth, metering.GasSchedule().BigIntAPICost.BigIntStorageLoadUnsigned, usedCache)
+
+	err = storage.UseGasForStorageLoad(bigIntStorageLoadUnsignedName, trieDepth, metering.GasSchedule().BigIntAPICost.BigIntStorageLoadUnsigned, usedCache)
+	if context.WithFault(err, runtime.BaseOpsErrorShouldFailExecution()) {
```

Comment:
runtime.BigIntAPIErrorShouldFailExecution() ?

Reply:
Will fix in an upcoming PR, the problem is rather widespread.